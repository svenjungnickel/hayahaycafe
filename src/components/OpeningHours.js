import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import { HTMLContent } from './content';

const OpeningHours = () => (
    <StaticQuery
        query={graphql`
            query {
                globalSettings: settingsYaml {
                    openingHours
                }
            }
        `}
        render={({ globalSettings: { openingHours } }) => (
            <>
                {!!openingHours && (
                    <>
                        <h2>Opening Hours</h2>
                        <HTMLContent content={openingHours} />
                    </>
                )}
            </>
        )}
    />
);

export default OpeningHours;
