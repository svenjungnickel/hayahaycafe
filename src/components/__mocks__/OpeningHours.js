import React from 'react';

const OpeningHours = () => <p>Opening Hours</p>;

export default OpeningHours;
