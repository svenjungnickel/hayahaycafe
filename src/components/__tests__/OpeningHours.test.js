import React from 'react';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';
import OpeningHours from '../OpeningHours';

describe('OpeningHours', () => {
    it('does not render empty data', () => {
        const data = {};
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <OpeningHours />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('does not render empty globalSettings data', () => {
        const data = {
            globalSettings: {},
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <OpeningHours />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly', () => {
        const data = {
            globalSettings: {
                openingHours: '<p>openingHours</p>',
            },
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <OpeningHours />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
