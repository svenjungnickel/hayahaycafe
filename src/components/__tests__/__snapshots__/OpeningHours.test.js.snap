// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningHours does not render empty data 1`] = `null`;

exports[`OpeningHours does not render empty globalSettings data 1`] = `null`;

exports[`OpeningHours renders correctly 1`] = `
<div
  className="openingHours"
>
  <h2>
    Opening Hours
  </h2>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": <p>
          openingHours
        </p>,
      }
    }
  />
</div>
`;
